### ISSUE TITLE
Implement a standard "controls grid" component for sidebar

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#43423]
## What problem does this address?

The controls in the sidebar should be laid out in a consistent grid, whether they are in `edit-post` (ToolsPanel) or `edit-site` (ad hoc layouts). Currently they each do their own styling for `rowGap`/`columnGap`, column count, and full-width items.

## What is your proposed solution?

Try making a standard layout component for this purpose, so they can be easily reused in either context. We can call it something like `ControlsGrid`.

**Comment by ciampo** [comment]
This could be an interesting exploration to inform some of the decision re. grid components discussed in https://github.com/WordPress/gutenberg/issues/43085

**Comment by jasmussen** [comment]
Here's a suggested grid: 

<img width="1026" alt="Grid" src="https://user-images.githubusercontent.com/1204802/187613848-7b278d22-3332-442b-8a5e-232e0b84002c.png">

Is that helpful?

**Comment by jasmussen** [comment]
I went ahead and created a separate issue for updating the default component sizes: #46734

**Comment by mirka** [comment]
Relevant: An implementation of the boundary-breaking buttons in the header labels #50855

**Comment by afercia** [comment]
Comine here from https://github.com/WordPress/gutenberg/pull/63806
Although this is somehow an old issue, I'd like to add a couple considerations:

Personally, I'm not sure the inconsistency in the placement of the 'Set custom size' toggle button helps with providing the highest level of usability and accessibility. Currently, this button is placed in a different spot depending on the control. Example screenshot:

![Screenshot 2024-07-25 at 08 37 49](https://github.com/user-attachments/assets/561bf2a1-8efa-4c37-81c4-38030580605f)

Sometimes the order is:
1. Label
2. Toggle
3. Control

Sometimes is:
1. Label
2. Control
3. Toggle

From a logical order perspective, I would expect this order:
1. Label: it tells me what the control is about and it makes sense to have this information first.
2. The control itself.
3. Lastly, the option to switch the control to an alternative version.

What I would like to see from the design team is an effort to make design that goes beyond providing a balanced, visually pleasant, grid. I would like to see a _functional_ analysis and consistency in the grouping and order of the controls.
